packages/actor-core/src/actor/config.ts

```diff
@@ -114,7 +114,7 @@ type CreateVars<S, CP, CS, V> =
 			/**
 			 * @experimental
 			 */
-			createVars: (c: ActorContext<S, CP, CS, undefined>) => V | Promise<V>;
+			createVars: (c: ActorContext<S, CP, CS, undefined>, driverCtx: unknown) => V | Promise<V>;
```

Comment:
I waffled on this a bit, but ultimately I think leaving the `driverCtx` as unknown is a decent tradeoff. The biggest problem is actors can be written to be agnostic and to be ran with different drivers. You could add a generic to the actor that would flow down to this, but there's no way to infer that. The best you could do is have a union of possible drivers you support. 

This PR leaves out a mechanism for determining what driver is currently being executed under. I still think we're going to want to figure that out, but I'd also rather not half bake an approach here that just ends up awkward in the future. 

For now this is a bit of an unsafe interface and is only for advanced users.

Reply:
I think `unknown` + passing it in as an arg is the right choice.

> You could add a generic to the actor that would flow down to this, but there's no way to infer that.

I think an acceptable option if we ever did this is to set the generic default to `unknown.`

> The best you could do is have a union of possible drivers you support.

Yep. I'd rather keep the `actor-core` package decoupled from the drivers. This diverges a bit from Hono's monolithic package, but ActorCore requires significantly more code to adapt to platforms.

> This PR leaves out a mechanism for determining what driver is currently being executed under.

I'm worried that whatever we decide would still result in a breaking change. Dropping a few thoughts on Discord in a moment.